### PR TITLE
Add workflow to verify installation of the plugin zip and dashboards …

### PR DESCRIPTION
**Pull request by derek-ho**
…comes up healthy

I have observed some issues within other plugins of issues being only caught at run time. This is because several things can go wrong during the build process, which may not be caught in a dev setup. This adds a workflow to verify that building and installing into OSD works on every PR.

Related issues:
https://github.com/opensearch-project/security-dashboards-plugin/issues/1709
https://github.com/opensearch-project/security-analytics-dashboards-plugin/pull/875
https://github.com/opensearch-project/dashboards-observability/pull/309
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5952

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/dashboards-notifications-test-and-build-workflow.yml [hunk 1]
```diff
@@ -129,7 +129,7 @@ jobs:
       - name: OpenSearch Dashboards Plugin Bootstrap
         run: |
           cd ./OpenSearch-Dashboards/plugins/dashboards-notifications
-          yarn osd bootstrap
+          yarn osd bootstrap --single-version=loose
```
Comment (derek-ho) [inline .github/workflows/dashboards-notifications-test-and-build-workflow.yml:132]: More context here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5561

**Review by SuZhou-Joe — APPROVED** [review]
LGTM.

File: .github/workflows/verify-binary-installation.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+name: 'Install Dashboards with Plugin via Binary'
+
+on: [push, pull_request]
+env:
+  OPENSEARCH_VERSION: '3.0.0'
```
Comment (SuZhou-Joe) [inline .github/workflows/verify-binary-installation.yml:5]: Seems this env variable is not needed?

**Comment by derek-ho** [comment]
```
curl: (7) Failed to connect to localhost port 5601 after 0 ms: Connection refused
{"type":"log","@timestamp":"2024-04-19T18:52:41Z","tags":["fatal","root"],"pid":2551,"message":"Error: Cannot find module '../../public/utils/constants'\nRequire stack:\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/configRoutes.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/index.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/plugin.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/index.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/src/core/server/
 FATAL  Error: Cannot find module '../../public/utils/constants'
Require stack:
- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/configRoutes.js

```

**Comment by derek-ho** [comment]
Seeing failures in the functional test repo, related to the notifications dashboard. Here is a PR run after I merge in main to this branch showing this new workflow would have caught this issue: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1146. @SuZhou-Joe @Hailong-am @xluo-aws can we get this merged to prevent future issues?

**Review by Hailong-am — APPROVED** [review]

**Comment by BionIT** [comment]
Thanks @derek-ho ! @riysaxen-amzn looks like the same issue has been captured, and now just need to fix. Having server side depends on the constants defined in public might be the cause, can you try moving to the common folder and see if it would pass this workflow Derek added?

**Pull request merged**